snudown.h

```diff
@@ -0,0 +1,118 @@
+/* snudown.h - header for snudown */
```

Comment:
sweet.  

AFAIK putting function definitions in `.h` is not best practice because it gets inlined everywhere in exec, but it should not matter here.  Probably someone who knows more about this code base should also weigh in on this decision.